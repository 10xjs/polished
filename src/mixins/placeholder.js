//@flow
function placeholder(styles: Object, parent: string = '&') {
  return {
    [`${parent}::-webkit-input-placeholder`]: {
      ...styles
    },
    [`${parent}:-moz-placeholder`]: {
      ...styles
    },
    [`${parent}::-moz-placeholder`]: {
      ...styles
    },
    [`${parent}:-ms-input-placeholder`]: {
      ...styles
    },
  }
}

export default placeholder
