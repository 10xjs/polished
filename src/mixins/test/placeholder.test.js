// @flow
import placeholder from '../placeholder'

describe('placeholder', function() {

  const styles = {
    'color': 'blue',
  }

  it('should properly pass styles object and parent', function() {
    expect({
      ...placeholder(styles,'input')
<<<<<<< HEAD
=======
    const styles = {
      'color': 'blue',
      'font-size': '12px',
    }

  it('should properly pass styles object and parent', function() {
    expect({
      input: {
        ...placeholder(styles,'input')
      }
>>>>>>> 088995e... chore(placeholder): Add placeholder mixin
=======
>>>>>>> 09e8acf... chore(all): More merge conflicts
    }).toMatchSnapshot()
  })

  it('should properly add rules when block has existing rules', function() {
    expect({
      'background': 'white',
      ...placeholder(styles,'input')
<<<<<<< HEAD
=======
      input: {
        'background': 'white',
        ...placeholder(styles,'input')
      }
>>>>>>> 088995e... chore(placeholder): Add placeholder mixin
=======
>>>>>>> 09e8acf... chore(all): More merge conflicts
    }).toMatchSnapshot()
  })

  it('should properly default to & when not passed a parent', function() {
    expect({
      input: {
        ...placeholder(styles)
      }
    }).toMatchSnapshot()
  })

})
