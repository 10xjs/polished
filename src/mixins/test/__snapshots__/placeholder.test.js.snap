exports[`placeholder should properly add rules when block has existing rules 1`] = `
Object {
<<<<<<< HEAD
  "background": "white",
  "input:-moz-placeholder": Object {
    "color": "blue",
  },
  "input:-ms-input-placeholder": Object {
    "color": "blue",
  },
  "input::-moz-placeholder": Object {
    "color": "blue",
  },
  "input::-webkit-input-placeholder": Object {
    "color": "blue",
=======
  "input": Object {
    "background": "white",
    "input:-moz-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "input:-ms-input-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "input::-moz-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "input::-webkit-input-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
>>>>>>> 088995e... chore(placeholder): Add placeholder mixin
  },
}
`;

exports[`placeholder should properly default to & when not passed a parent 1`] = `
Object {
  "input": Object {
    "&:-moz-placeholder": Object {
      "color": "blue",
<<<<<<< HEAD
    },
    "&:-ms-input-placeholder": Object {
      "color": "blue",
    },
    "&::-moz-placeholder": Object {
      "color": "blue",
    },
    "&::-webkit-input-placeholder": Object {
      "color": "blue",
=======
      "font-size": "12px",
    },
    "&:-ms-input-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "&::-moz-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "&::-webkit-input-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
>>>>>>> 088995e... chore(placeholder): Add placeholder mixin
    },
  },
}
`;

exports[`placeholder should properly pass styles object and parent 1`] = `
Object {
<<<<<<< HEAD
  "input:-moz-placeholder": Object {
    "color": "blue",
  },
  "input:-ms-input-placeholder": Object {
    "color": "blue",
  },
  "input::-moz-placeholder": Object {
    "color": "blue",
  },
  "input::-webkit-input-placeholder": Object {
    "color": "blue",
=======
  "input": Object {
    "input:-moz-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "input:-ms-input-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "input::-moz-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
    "input::-webkit-input-placeholder": Object {
      "color": "blue",
      "font-size": "12px",
    },
>>>>>>> 088995e... chore(placeholder): Add placeholder mixin
  },
}
`;
